src/io/fasta.rs

```diff
@@ -3,14 +3,100 @@
 // This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! FASTA format reading and writing.
+//! Structs and trait to read and write file in FASTA format.
 //!
 //! # Example
 //!
+//! ## Read
+//!
+//! In this example we parse fasta file from stdin and compute some statistics
+//!
+//! ```
+//! // import functions (at top of script)
+//! use std::io;
+//! use bio::io::fasta;
+//! // call within functions
+//! let mut reader = fasta::Reader::new(io::stdin());
+//!
+//! let mut nb_reads = 0;
+//! let mut nb_bases = 0;
+//!
+//! for result in reader.records() {
+//!     let record = result.expect("Error durring fasta record parsing");
+//!     println!("{}", record.id());
+//!
+//!     nb_reads += 1;
+//!     nb_bases += record.seq().len();
+//! }
+//!
+//! println!("Number of reads: {}", nb_reads);
+//! println!("Number of bases: {}", nb_bases);
+//! ```
+//!
+//! We can also use a `while` loop to iterate over record
```

Comment:
record -> records

Might also be good to mention that the `while` loop method is *slightly* faster than the `for` loop method.

Reply:
```suggestion
//! We can also use a `while` loop to iterate over records.
//! This is slightly faster than the `for` loop.
```